src/app/api/crud/accesses.py

```diff
@@ -0,0 +1,99 @@
+    """Insert an access entry in the accesses table, call within a transaction to reuse returned access id."""
+    await check_login_existence(accesses, login)
+
+    # Hash the password
+    pwd = await security.hash_password(password)
+
+    access = AccessCreation(login=login, hashed_password=pwd, scopes=scopes)
+    entry = await base.create_entry(accesses, access)
+
+    return AccessRead(**entry)
+
+
+async def update_access_pwd(accesses: Table, payload: Cred, access_id: int) -> None:
+    """Update the access password using provided access_id."""
+    # Update the access entry with the hashed password
+    updated_payload = CredHash(hashed_password=await security.hash_password(payload.password))
+
+    await base.update_entry(accesses, updated_payload, access_id)  # update & check if access_id exists
+
+
+async def create_accessed_entry(
+    table: Table,
+    accesses: Table,
+    payload: Union[UserAuth, DeviceAuth],
+    schema: Type[Union[UserCreation, DeviceCreation]],
+) -> Dict:
+    """Create an access and refers it to a new entry (User, Device, ...)."""
+    # Ensure database consistency between tables with a transaction
+    async with base.database.transaction():
+        access_entry = await post_access(accesses, payload.login, payload.password, payload.scopes)
+        entry = await base.create_entry(table, schema(**payload.dict(), access_id=access_entry.id))
+
+    return entry
+
+
+async def update_accessed_entry(table: Table, accesses: Table, entry_id: int, payload: Any):
+    """Update an entry with a special treatment regarding login: if login is set -> update corresponding access."""
+    # Ensure database consistency between tables with a transaction (login must remain the same in table & accesses)
+    async with base.database.transaction():
+
+        # Handle access update only if login is set
+        if payload.login is not None:
+
+            # Need to retrieve access_id from entry
+            origin_entry = await base.get_entry(table, entry_id)  # assert entry exist
+
+            # Update corresponding access only if login need to change
+            if payload.login != origin_entry["login"]:
+                await check_login_existence(accesses, payload.login)  # assert new login does not exist in accesses
+                await update_login(accesses, payload.login, origin_entry["access_id"])
+
+        # Update entry with input payload
+        entry = await base.update_entry(table, payload, entry_id)
+
+    return entry
+
+
+async def delete_accessed_entry(table: Table, accesses: Table, entry_id: int):
+    """Delete an entry (User, Device, ...) and the corresponding access."""
+    # Ensure database consistency between tables with a transaction
+    async with base.database.transaction():
+        entry = await base.delete_entry(table, entry_id)
+        await base.delete_entry(accesses, entry["access_id"])
+
+    return entry
```

Comment:
Do we have to regroup the table + its access operation in the same function for the db transaction to be used?
It's better if we can tell easily what's going on from the route code when possible

Reply:
I am not sure to understand your question haha

Reply:
Sorry :sweat_smile: 
I meant, if possible, having 2 separate instructions in the route code for the route table operation, and for its access table operation would be nice. Here you regrouped both ops in a function called by the route, which in the end, means that we don't see in the route code which tables get modified

Reply:
Ok I see, that's what I planned to do with this db.transaction decorator on each route but I had some issues during the tests and it was not a good practice (a transaction must be closed asap) and then I figured out that there was a lot of code duplication between users & devices: putting the same operations in single functions with some parameters should make it easier to maintain. At least we have the tables in argument (users or devices and accesses) so we know what tables are involved from routes !

Reply:
yes, I was just nitpicking! We can easily think about a refactoring option later!